homeworks/week02/README.md

```diff
@@ -0,0 +1,232 @@
+  object
+  2. Concatenate the return value from this function call to your new
+  string variable.
+
+3. Return this new string variable, as your function's return value
+
+If you reload your page, you should now see a list displayed of all
+movies, with whatever styling you have added for each different genre.
+
+Again, commit your code with a helpful comment before proceeding.
+
+## Step 3:
+
+Now we'll demonstrate the power of functions. Without changing your
+```render()``` function, you'll change the way that the list of movies
+gets rendered into HTML. To do this, you'll create a new function that
+creates HTML differently, so the looping code that you just wrote in
+Step 2 won't need to change, except that you'll call this new function
+instead, and you'll see a very different result.
+
+The goal will be to add a new, more compact way of rendering each
+movie, kind of like the "mobile version" of the movie list.
+
+To do this, add a new function that implements the following
+signature:
+
+```javascript
+function renderMovieCompact(movie)
+```
+
+Just like in Step 1, this function should use string concatenation and
+all the attributes of a single object (from the ```movie``` argument)
+to produce a single string of HTML, which the function should then
+return.
+
+The only difference is that now you will create a more compact
+```<li>``` rather than a ```<div>```. So the same movie object from
+Step 1 should now produce this HTML:
+
+```html
+<li class="movie animation">
+  <span class="name">Minions</span>
+  (<span class="year">2015</span>,
+  <span class="genre"> animation</span>)
+  starring: Sandra Bullock, Jon Hamm, Michael Keaton
+</li>
+```
+
+Modify ```render()``` so that instead of calling ```renderMovie()```,
+it calls ```renderMovieCompact()``` instead. If you reload the page,
+you should now see a more compact movie list.
+
+As always, commit your code with a helpful comment before proceeding.
+
+## Step 4
+
+Finally, we'll put everything together.
+
+Modify the body of your loop in ```render()``` by adding an ```if```
+statement, so that if ```displayWide``` is ```true```, you call
+```renderMovie()```, otherwise, call ```renderMovieCompact()```.
+
+Now, if you resize your browser window (to less than 500px) and
+reload, you should see the compact version, but if your browser is
+wider, you should see the full version.
+
+You should also add some additional CSS code. First, before your loop
+in ```render()```, add another ```if``` statement so that if
+```displayWide``` is ```true```, you put a ```<div>``` with class
+```big```, otherwise put a ```<div>``` with class ```small``` and a
+```<ul>```. Add an ```if``` statement with the corresponding close
+tags after your loop. Now you can add some CSS to style the different
+display versions accordingly. Here is a snippet to get you started:
+
+```
+.big {
+    width: 80%;
+    margin: 25px auto;
+}
+.small { width: 100%; }
```

Comment:
Might be confusing to write the whole block on one line, if this wasn't covered in class. Yes, it's equivalent, but if you've never seen it before, it might be a puzzle to figure out since it looks different.